scripts/src/java/org/oppia/android/scripts/SyntaxErrorHandler.kt

```diff
@@ -0,0 +1,41 @@
+package org.oppia.android.scripts
+
+import org.xml.sax.ErrorHandler
+import org.xml.sax.SAXParseException
+
+/**
+ * Class for custom error handling of the parse exception thrown
+ * by the parser, to log the failure message in a cleaner way.
+ */
+class SyntaxErrorHandler(val repoPath: String) : ErrorHandler {
+  override fun warning(e: SAXParseException) {
+    logCheckFailure(e, repoPath)
+  }
+
+  override fun error(e: SAXParseException) {
+    logCheckFailure(e, repoPath)
+  }
+
+  override fun fatalError(e: SAXParseException) {
+    logCheckFailure(e, repoPath)
+  }
+
+  /**
+   * Logs the failures for XML syntax validation.
+   *
+   * @param e the parsing exception thrown by the parser
+   * @param repoPath the path of the repo to be analyzed
+   * @return log the failure
+   */
+  private fun logCheckFailure(parseException: SAXParseException, repoPath: String) {
+    val failureMessage =
```

Comment:
I suggest formatting this message similarly to the version suggested in #3340.

Reply:
Done.